docs/development/design/automation-rules.rst

```diff
@@ -0,0 +1,101 @@
+- Automation rules are currently matched against new versions,
+  but they can also be executed over deleted versions (https://github.com/readthedocs/readthedocs.org/pull/7644).
+
+Improvements
+------------
+
+Keep track of the ``n`` latest matches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We do this with webhook exchanges (keeping the latest 10)
+https://github.com/readthedocs/readthedocs.org/blob/f5b76e87a3b970d42228f5972ae9a50364c3c76c/readthedocs/integrations/models.py#L99-L111.
+See https://github.com/readthedocs/readthedocs.org/issues/6393.
+
+Live feedback of versions that match a rule
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When creating/editing a rule, it would be useful to know what versions are going to match.
+We could list the current versions that match and/or allow the user to enter a custom input.
+A new enpoint in api v2/v3 should be added.
+
+Easy way to invert a match
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, users need to write a negative look ahead,
+which isn't really user-friendly or easy to remember/write.
+Having an explicit option to invert a match would be more useful,
+see https://github.com/readthedocs/readthedocs.org/issues/6354.
+
+New features
+------------
+
+Change the privacy level of external versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Read the Docs for business all versions are private by default.
+Some users may want to change the privacy level of external versions.
+See https://github.com/readthedocs/readthedocs-corporate/issues/1063.
+
+If automation rules are going to be used, users will need to be able to edit those versions.
+This means having a place to list all external versions and allow to edit some properties,
+privacy level is the only option that makes sense maybe also allow to de-activate it so docs aren't accessible.
+
+Decide what PRs should be built
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use automation rules to decide which pull requests should be built,
+this is for example only built branches that are going to be merged against an specific branch.
+Or branch names that match a pattern, like (``^docs/.*``).
+
+.. note::
+
+   Here we don't need a separate UI element yet,
+   as users can still trigger a new build by pushing to that branch if they got the pattern wrong.
+
+Change the slug a version served from
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, we create the version slug from the version name, and this can't be changed.
+Some users may end up with a slug they didn't intend,
+Or there are users that want their slug to be different from the name of the branch/tag,
+such as https://github.com/readthedocs/readthedocs.org/issues/7078.
+
+The models from automation rules already support passing enxtra arguments to be used by an action.
+Before implementing this rule, we need to allow adding an alias for versions manually,
+this would be by creating an alias (should the version be served from two slugs?)
```

Comment:
I think it would be good to find a way to automatically create redirects on slug changes, so we don't break old links.

Reply:
yeah, that was pointed in the design doc of changing the slug https://github.com/readthedocs/readthedocs.org/pull/6273/files#diff-9561f5273ac1998bbd574fc850e2c100688ed8b291273f690fd49d6df45639beR80-R82